### ISSUE TITLE
Bug in SpListPresenter: selection not reset upon replacement of items

### ISSUE THREAD

**Issue by stacksmith** [pharo-spec/Spec#841]
I am building a navigation UI which shows a list of destinations for the user to click on; upon a click a new list is shoved into the ListPresenter.  
Since the only way to react to changes is by storing a `whenSelectionChangedDo:` block, I assumed that `resetListSelection` would deselect the old selection.  However, it does not.  
When the list shows a **single item**, it remains selected [edit: if the click was on the first item].  **Clicking** on the new list item (which is for some reason still selected) **now does nothing** as there is no way to effect selection change with a single selected item,
and my navigation system is not responsive.

```
'From Pharo9.0.0 of 14 February 2020 [Build information: Pharo-9.0.0+build.66.sha.143ec632296ac259948b035877e79236376cffdd (64 Bit)] on 25 February 2020 at 1:48:52.126709 pm'!
SpPresenter subclass: #SpBuggy
	slots: { #list }
	classVariables: {  }
	package: 'SpBuggery'!

!SpBuggy methodsFor: 'initialization' stamp: 'StackSmith 2/25/2020 13:47'!
connectPresenters
	list whenSelectionChangedDo: [ 
		Transcript show: 'selection changed.'; cr.
		list items: { 'newItem' }.
		list resetListSelection].! !

!SpBuggy methodsFor: 'initialization' stamp: 'StackSmith 2/25/2020 13:48'!
initializePresenters
	list := self newList
		items: { 'click for bug' . 'fine here'};
		display: #yourself.
! !

"-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- "!

SpBuggy class
	slots: {  }!

!SpBuggy class methodsFor: 'specs' stamp: 'StackSmith 2/25/2020 13:41'!
example
	<example>
	| presenter |
	presenter := self new.
	presenter openWithSpec extent: 900 @ 400.
	^ presenter! !

!SpBuggy class methodsFor: 'specs' stamp: 'StackSmith 2/25/2020 13:44'!
defaultSpec 	
	^ SpBoxLayout newHorizontal
		add: #list   expand: false;
		yourself! !
```

**Comment by stacksmith** [comment]
Note that clicking on 'click for bug', item #1 results in item#1 of the new list to be selected.  Clicking on item #2 works fine.  Hope that helps.

**Comment by estebanlm** [comment]
this is a duplication of #850 and is fixed in #852 :)
